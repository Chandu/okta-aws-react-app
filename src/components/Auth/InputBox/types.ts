import { ChangeEvent } from 'react';

export type Props = {
    type: string;
    name?: string;
    value: string;
    label: string;
    errorText?: string;
    fullWidth?: boolean;
    placeholderText: string;
    onChange: (e: ChangeEvent<HTMLInputElement>) => void;
};
