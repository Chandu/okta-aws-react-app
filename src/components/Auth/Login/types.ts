export type Props = {
    toggleWidget: () => void;
};
