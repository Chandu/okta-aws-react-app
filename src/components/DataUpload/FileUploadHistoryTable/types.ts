export type Props = {
    fileList: Array<{ [key: string]: any }>;
    isLoading: boolean;
};
