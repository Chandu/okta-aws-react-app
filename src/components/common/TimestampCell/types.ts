import { Table } from '@medly-components/core';
import { GetComponentProps } from '@medly-components/utils';

export type ComponentType = GetComponentProps<typeof Table>['columns'][0]['component'];
