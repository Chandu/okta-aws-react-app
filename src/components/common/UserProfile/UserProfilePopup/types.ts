export interface Props {
    signOut: () => void;
    userName: string;
    showSeparator: boolean;
}
