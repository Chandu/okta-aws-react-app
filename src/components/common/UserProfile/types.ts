export interface Props {
    showSeparator?: boolean;
}
