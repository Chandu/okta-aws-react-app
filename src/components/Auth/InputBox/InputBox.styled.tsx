import { Text } from '@medly-components/core';
import { HTMLProps } from '@medly-components/utils';
import { primaryTheme } from '@theme';
import { css, styled } from '@utils';

export const InputWrapper = styled('div')<{ fullWidth?: boolean }>`
    ${({ fullWidth }) =>
        fullWidth &&
        css`
            width: 100%;
            & {
                flex-grow: 1;
            }
        `};
`;

export const Label = styled(Text)`
    display: block;
    color: ${({ theme }) => theme.colors.white};
`;

export const InputBox = styled('input')<{ fullWidth?: boolean } & HTMLProps<HTMLInputElement>>`
    border: none;
    margin: 0.5rem 0;
    font-size: 1.4rem;
    line-height: 2rem;
    border-radius: 0.4rem;
    padding: 1.3rem 1.6rem;
    box-sizing: border-box;
    ${({ fullWidth }) =>
        fullWidth &&
        css`
            width: 100%;
        `}
    ::placeholder {
        color: ${primaryTheme.colors.grey[500]};
    }
`;
