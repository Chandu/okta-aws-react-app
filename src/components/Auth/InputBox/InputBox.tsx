import { Box } from '@medly-components/core';
import { defaultTheme } from '@theme';
import { ThemeProvider, updateNestedValue } from '@utils';
import { FC, memo } from 'react';
import * as Styled from './InputBox.styled';
import { Props } from './types';

export const InputBox: FC<Props> = memo(({ name, placeholderText, onChange, value, label, type, fullWidth, errorText }) => {
    return (
        <Styled.InputWrapper fullWidth={fullWidth}>
            <ThemeProvider
                theme={updateNestedValue(defaultTheme, 'input', {
                    ...defaultTheme.input,
                    placeHolderColor: defaultTheme.colors.grey[400]
                })}
            >
                <Styled.Label textVariant="h5">{label}</Styled.Label>
                <Styled.InputBox
                    id={`${name}-id`}
                    key={name}
                    {...{ name, onChange, value, type, fullWidth }}
                    placeholder={placeholderText}
                    required
                />
                {errorText && (
                    <Box
                        my={-2}
                        py={0}
                        color={defaultTheme.colors.red[500]}
                        borderColor={defaultTheme.colors.red[500]}
                        borderRadius={2}
                        borderWidth={1}
                    >
                        {errorText}
                    </Box>
                )}
            </ThemeProvider>
        </Styled.InputWrapper>
    );
});

InputBox.displayName = 'InputBox';
export default InputBox;
