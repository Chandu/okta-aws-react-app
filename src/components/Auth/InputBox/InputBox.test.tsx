import { fireEvent, render, screen } from '@utils/test-utils';
import React from 'react';
import { InputBox } from './InputBox';
import { Props } from './types';

const mockOnChange = jest.fn(),
    renderer = (props?: Partial<Props>) =>
        render(
            <InputBox
                type="text"
                name="input"
                onChange={mockOnChange}
                placeholderText="input box"
                value="some value"
                label="input label"
                {...props}
            />
        );

describe('InputBox', () => {
    it('should render properly', () => {
        const { container } = renderer();
        expect(container).toMatchSnapshot();
    });

    it('should render properly with full width', () => {
        const { container } = renderer({ fullWidth: true });
        expect(container).toMatchSnapshot();
    });

    it('should call mockOnChange', () => {
        renderer();
        fireEvent.change(screen.getByPlaceholderText('input box'), {
            target: { value: 'new value' }
        });
        expect(mockOnChange).toBeCalledTimes(1);
    });

    it('should show error message on validator returning true', () => {
        renderer({ errorText: 'Field validation error' });
        expect(screen.getByText('Field validation error')).toBeInTheDocument();
    });
});
