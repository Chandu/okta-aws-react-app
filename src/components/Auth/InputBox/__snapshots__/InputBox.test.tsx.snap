// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBox should render properly 1`] = `
.c0 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 500;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  line-height: 2rem;
  text-align: initial;
}

.c1 {
  display: block;
  color: #ffffff;
}

.c2 {
  border: none;
  margin: 0.5rem 0;
  font-size: 1.4rem;
  line-height: 2rem;
  border-radius: 0.4rem;
  padding: 1.3rem 1.6rem;
  box-sizing: border-box;
}

.c2::-webkit-input-placeholder {
  color: #98A7B7;
}

.c2::-moz-placeholder {
  color: #98A7B7;
}

.c2:-ms-input-placeholder {
  color: #98A7B7;
}

.c2::placeholder {
  color: #98A7B7;
}

<div>
  <div
    class=""
  >
    <h5
      class="c0 c1"
    >
      input label
    </h5>
    <input
      class="c2"
      id="input-id"
      name="input"
      placeholder="input box"
      required=""
      type="text"
      value="some value"
    />
  </div>
</div>
`;

exports[`InputBox should render properly with full width 1`] = `
.c1 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 500;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  line-height: 2rem;
  text-align: initial;
}

.c0 {
  width: 100%;
}

.c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c2 {
  display: block;
  color: #ffffff;
}

.c3 {
  border: none;
  margin: 0.5rem 0;
  font-size: 1.4rem;
  line-height: 2rem;
  border-radius: 0.4rem;
  padding: 1.3rem 1.6rem;
  box-sizing: border-box;
  width: 100%;
}

.c3::-webkit-input-placeholder {
  color: #98A7B7;
}

.c3::-moz-placeholder {
  color: #98A7B7;
}

.c3:-ms-input-placeholder {
  color: #98A7B7;
}

.c3::placeholder {
  color: #98A7B7;
}

<div>
  <div
    class="c0"
  >
    <h5
      class="c1 c2"
    >
      input label
    </h5>
    <input
      class="c3"
      id="input-id"
      name="input"
      placeholder="input box"
      required=""
      type="text"
      value="some value"
    />
  </div>
</div>
`;
